File: src/java/Trees/PostOrderTraversal.java
public class PostOrderTraversal{
  public void postorderTraversal(TreeNode root) {
        if (root == null) {
            return;
        }
        postorderTraversal(root.left);  // Traverse left subtree
        postorderTraversal(root.right);  // Traverse right subtree
        System.out.print(root.value + " ");  // Process the root
    }

    public static void main(String[] args) {
        TreeNode root = new TreeNode(1);
        root.left = new TreeNode(2);
        root.right = new TreeNode(3);
        root.left.left = new TreeNode(4);
        root.left.right = new TreeNode(5);

        BinaryTree tree = new BinaryTree();
        tree.postorderTraversal(root);
    }
}

File: src/python/Trees/PostOrderTraversal.py
class TreeNode:
    def __init__(self, value):
        self.value = value
        self.left = None
        self.right = None

def postorder_traversal(root):
    if root:
        postorder_traversal(root.left)  # Traverse left subtree
        postorder_traversal(root.right)  # Traverse right subtree
        print(root.value, end=' ')  # Process the root

if __name__ == '__main__':
    root = TreeNode(1)
    root.left = TreeNode(2)
    root.right = TreeNode(3)
    root.left.left = TreeNode(4)
    root.left.right = TreeNode(5)

    postorder_traversal(root)
